File: src/main/java/com/clearcont/clearcontapp/model/Empresa.java
package com.clearcont.clearcontapp.model;

import com.clearcont.clearcontapp.helpers.CNPJ;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.Email;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Empresa {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nomeEmpresa;
    @CNPJ
    private String cnpj;
    @Email
    private String email;
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/models/Empresa.kt
package br.com.clearcont.clearcontwebapp.models

import jakarta.persistence.*
import jakarta.validation.constraints.Email
import org.hibernate.validator.constraints.br.CNPJ

@Entity

class Empresa(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @Column(length = 50)
    var nomeEmpresa: String? = null,
    @CNPJ
    var cnpj: String? = null,
    var email: @Email String? = null
)
